File: frontend/packages/types/src/track.ts
export interface Track {
  id: string;
  name: string;
  code: string;
  slug?: string;
  description?: string;
  is_active?: boolean;
  module_count?: number;
  fee_inr?: number;
  duration?: string;
  highlights?: string;
  outcomes?: string;
}

export interface TrackDetail extends Track {}

File: backend/app/models/track.py
from beanie import Document
from typing import Optional
from datetime import datetime


class Track(Document):
    name: str
    code: str
    description: Optional[str] = None
    is_active: bool = True
    module_count: int = 0
    created_at: datetime = datetime.utcnow()
    updated_at: datetime = datetime.utcnow()

    class Settings:
        name = "tracks"
